src/main/java/io/debezium/connector/jdbc/JdbcChangeEventSink.java

```diff
@@ -127,7 +127,14 @@ public void execute(Collection<SinkRecord> records) {
                     flushBuffer(tableId, updateBufferByTable.get(tableId).flush());
                 }
 
-                RecordBuffer tableIdBuffer = deleteBufferByTable.computeIfAbsent(tableId, k -> new RecordBuffer(config));
+                Buffer tableIdBuffer;
+                if (config.isUseReductionBuffer()) {
+                    tableIdBuffer = deleteBufferByTable.computeIfAbsent(tableId, k -> new ReducedRecordBuffer(config));
+                }
+                else {
+                    tableIdBuffer = deleteBufferByTable.computeIfAbsent(tableId, k -> new RecordBuffer(config));
+                }
+
```

Comment:
Let's move this to a separate method to centralize the logic, i.e.:
```java
private Buffer resolveBuffer(Map<TableId, Buffer> bufferMap, TableId tableId) {
    if (config.isUseReductionBuffer()) {
        return bufferMap.computeIfAbsent(tableId, k -> new ReducedRecordBuffer(config));
    }
    else {
        return bufferMap.computeIfAbsent(tableId, k -> new RecordBuffer(config));
    }
}
```
Then at the two call locations, this can be simplified to:
```java
Buffer tableIdBuffer = resolveBuffer(deleteBufferByTable, tableId);
```
or
```java
Buffer tableIdBuffer = resolveBuffer(updateBufferByTable, tableId);
```